internal/querycoord/channel_allocator.go

```diff
@@ -31,13 +33,17 @@ func defaultChannelAllocatePolicy() ChannelAllocatePolicy {
 }
 
 // ChannelAllocatePolicy helper function definition to allocate dmChannel to queryNode
-type ChannelAllocatePolicy func(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, wait bool, excludeNodeIDs []int64) error
+type ChannelAllocatePolicy func(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, metaCache Meta, wait bool, excludeNodeIDs []int64) error
 
-func shuffleChannelsToQueryNode(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, wait bool, excludeNodeIDs []int64) error {
+func shuffleChannelsToQueryNode(ctx context.Context, reqs []*querypb.WatchDmChannelsRequest, cluster Cluster, metaCache Meta, wait bool, excludeNodeIDs []int64) error {
+	if len(reqs) == 0 {
+		return nil
+	}
 	for {
-		availableNodes, err := cluster.onlineNodes()
-		if err != nil {
-			log.Debug(err.Error())
+		availableNodes := cluster.onlineNodes()
+		if len(availableNodes) == 0 {
```

Comment:
need another log, print out onlineNodes, excludeNodes and there channel infos.

Reply:
fixed